From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 25 Jul 2022 00:22:00 +0200
Subject: [PATCH] Don't bundle fonts FFS!

There's no reason to bundle fonts in the binary, we install them separately.
These fonts have 15 MiB in total!

Roboto is a sans-serif font, Wezterm uses it for its UI; any other sans-serif
font will work.

LastResortHE is a special-purpose font which may cause troubles when installed
as a normal font (https://github.com/unicode-org/last-resort-font/issues/3).
For some obscure reason, WezTerm may panic if this font is not available
(https://github.com/wez/wezterm/pull/2305#issuecomment-1194779072).

--- a/wezterm-font/src/parser.rs
+++ b/wezterm-font/src/parser.rs
@@ -690,7 +690,9 @@
         };
     }
     let lib = crate::ftwrap::Library::new()?;
-    for (data, name) in &[
+
+    #[cfg(test)]
+    let fonts = &[
         font!("../../assets/fonts/JetBrainsMono-BoldItalic.ttf"),
         font!("../../assets/fonts/JetBrainsMono-Bold.ttf"),
         font!("../../assets/fonts/JetBrainsMono-ExtraBoldItalic.ttf"),
@@ -720,7 +722,13 @@
         font!("../../assets/fonts/NotoColorEmoji.ttf"),
         font!("../../assets/fonts/Symbols-Nerd-Font-Mono.ttf"),
         font!("../../assets/fonts/LastResortHE-Regular.ttf"),
-    ] {
+    ];
+    #[cfg(not(test))]
+    let fonts = &[
+        font!("../../assets/fonts/LastResortHE-Regular.ttf"),
+    ];
+
+    for (data, name) in fonts {
         let locator = FontDataHandle {
             source: FontDataSource::BuiltIn { data, name },
             index: 0,
